### DISCUSSION TITLE
how to find the actual versions of installed packages which are specified as "latest" in package.json with bun?

### DISCUSSION THREAD

**Discussion by MonaAghili** [oven-sh/bun#10181] [General]
I created my `Next.js` application using Bun.
I changed my project's `Next.js` dependency to the `latest` by setting it to `"next": "latest"` in the `package.json` file.
Now, I'd like to determine the versions of the installed packages, I tried `bun bun.lock` but the command didn't list them in the terminal.

**Comment by Jarred-Sumner** [comment]
`bun pm ls`
<img width="362" alt="image" src="https://github.com/oven-sh/bun/assets/709451/36edb49f-2d84-4888-93cd-ac1fe1960245">

**Reply by MonaAghili** [reply]
Thank you

**Reply by lgarron** [reply]
Is there a version of this that is machine-parseable? `npm` has `npm ls --json` but `bun pm ls --json` doesn't seem to produce different output.

**Comment by barclayd** [comment]
Does this work for bun workspaces too please?
Edit: Looks like `bun pm ls --all` does the trick

**Comment by mariansam** [comment]
npm supports `npm ls package-name` to show only installed versions of `package-name` (and the reason they're installed), `bun pm ls` doesn't support that yet :(

**Reply by GoodPie** [reply]
Just in case you weren't aware, you can use `grep` to filter the output which would give you something similar to `npm ls package-name`. Missing some functionality but it's a nice work-around in the mean time.

```bash
bun pm ls | grep package-name
```
